common/lib/xmodule/xmodule/modulestore/xml_exporter.py

```diff
@@ -122,72 +192,54 @@ def export_to_xml(modulestore, contentstore, course_key, root_dir, course_dir):
             policy = {'course/' + course.location.name: own_metadata(course)}
             course_policy.write(dumps(policy, cls=EdxJSONEncoder, sort_keys=True, indent=4))
 
-        #### DRAFTS ####
         # xml backed courses don't support drafts!
         if course.runtime.modulestore.get_modulestore_type() != ModuleStoreEnum.Type.xml:
-            # NOTE: we need to explicitly implement the logic for setting the vertical's parent
-            # and index here since the XML modulestore cannot load draft modules
-            with modulestore.branch_setting(ModuleStoreEnum.Branch.draft_preferred, course_key):
-                draft_modules = modulestore.get_items(
-                    course_key,
-                    qualifiers={'category': {'$nin': DIRECT_ONLY_CATEGORIES}},
-                    revision=ModuleStoreEnum.RevisionOption.draft_only
-                )
-
-                if draft_modules:
-                    draft_course_dir = export_fs.makeopendir(DRAFT_DIR)
+            _export_drafts(modulestore, course_key, export_fs, xml_centric_course_key)
 
-                    # accumulate tuples of draft_modules and their parents in
-                    # this list:
-                    draft_node_list = []
 
-                    for draft_module in draft_modules:
-                        parent_loc = modulestore.get_parent_location(
-                            draft_module.location,
-                            revision=ModuleStoreEnum.RevisionOption.draft_preferred
-                        )
-
-                        # if module has no parent, set its parent_url to `None`
-                        parent_url = None
-                        if parent_loc is not None:
-                            parent_url = parent_loc.to_deprecated_string()
-
-                        draft_node = draft_node_constructor(
-                            draft_module,
-                            location=draft_module.location,
-                            url=draft_module.location.to_deprecated_string(),
-                            parent_location=parent_loc,
-                            parent_url=parent_url,
-                        )
-
-                        draft_node_list.append(draft_node)
+def export_library_to_xml(modulestore, contentstore, library_key, root_dir, library_dir):
```

Comment:
I'm surprised at the amount of differing code between libraries and courses during export. I expected libraries to not have the extra content, but it seems like the rest of the data should be the same.